src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -0,0 +1,261 @@
+
+			plugin.debugMessage("Preparing statement for update.");
+			statement.executeUpdate();
+			plugin.debugMessage("Successfully executed update statement. (" + sql + ")");
+
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute update statement. (" + sql + ")");
+			e.printStackTrace();
+		}
+
+	}
+
+	public ResultSet query(final String sql) {
+
+		try (Connection connection = this.getConnection()){
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+
+			plugin.debugMessage("Preparing statement for query.");
+			ResultSet resultSet = statement.executeQuery(sql);
+			plugin.debugMessage("Successfully executed query statement. (" + sql + ")");
+
+			return resultSet;
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute query statement. (" + sql + ")");
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	public void setUpTable() {
+		this.connect();
+
+		plugin.debugMessage("Attempting to set up tables if they do not exist.");
+		final String update = "CREATE TABLE IF NOT EXISTS itemslogger("
+				+ "uuid VARCHAR(36) NOT NULL, "
+				+ "inventory VARCHAR(12000), "
+				+ "cause VARCHAR(255) NOT NULL, "
+				+ "loc_x REAL NOT NULL, "
+				+ "loc_y REAL NOT NULL, "
+				+ "loc_z REAL NOT NULL, "
+				+ "experience REAL NOT NULL, "
+				+ "date DATE NOT NULL, "
+				+ "time TIME NOT NULL);";
+		this.update(update);
+	}
+
+	private ItemPlayer initializePlayer(ResultSet rs) {
+
+		try {
+			UUID uuid = UUID.fromString(rs.getString("uuid"));
+			String inv = rs.getString("inventory");
+			String cause = rs.getString("cause");
+			int x = rs.getInt("loc_x");
+			int y = rs.getInt("loc_y");
+			int z = rs.getInt("loc_z");
+			int experience = rs.getInt("experience");
+			Date date = rs.getDate("date");
+			Time time = rs.getTime("time");
+
+			return new ItemPlayer(uuid, inv, cause, x, y, z, experience, date, time);
+		} catch (SQLException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	public CompletableFuture<ItemPlayer[]> retrieveListAsync(UUID uuid, String date, String cause, int max, int min) {
+
+		return CompletableFuture.supplyAsync(() -> {
+			String sql = " SELECT * FROM itemslogger WHERE uuid = '" + uuid + "'";
+
+			if (cause != null) {
+				sql += " AND cause = '" + cause + "'";
+			}
+
+			if (date != null) {
+				sql += " AND date = '" + date + "'";
+			}
+
+			sql += " ORDER BY date DESC, time DESC";
+			ResultSet rs = storage.query(sql);
+
+			try {
+				List<ItemPlayer>playerDataArray = new ArrayList<ItemPlayer>();
+				int i = 0;
+				while (rs.next()) {
+					if ((i<max) && (i >= min)) {
+						playerDataArray.add(initializePlayer(rs));
+					}
+					i++;
+				}
+				return playerDataArray.toArray(new ItemPlayer[0]);
+			} catch (SQLException e) {
+				e.printStackTrace();
+				return null;
+			}
+		});
+
+	}
+
+	public ItemPlayer[] retrieveList(UUID uuid, String date, String cause, Integer index) {
```

Comment:
There are very few reasons to ever use the object wrappers for atomic values
```suggestion
	public ItemPlayer[] retrieveList(UUID uuid, String date, String cause, int index) {
```